[CI] HoistRemoteEnrichTopNTests testTopNSortExpressionWithinRemoteEnrichAliasing failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #11321 / windows-2025_checkpart3_platform-support-windows](https://gradle-enterprise.elastic.co/s/fjt3q5s5g7b4w)
- [elasticsearch-periodic #11341 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/zl2awxvnryazk)
- [elasticsearch-periodic-platform-support #11321 / windows-2022_checkpart3_platform-support-windows](https://gradle-enterprise.elastic.co/s/pw7wsadputemq)
- [elasticsearch-periodic #11341 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/ssypozk7ny3mi)
- [elasticsearch-periodic-platform-support #11321 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/3j7uqu5k7pypw)
- [elasticsearch-periodic-platform-support #11321 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/eud36vvihtyq2)
- [elasticsearch-periodic-platform-support #11321 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/fsoztufmsf3no)
- [elasticsearch-periodic-platform-support #11321 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/teziqs4jtwfo6)
- [elasticsearch-periodic-platform-support #11321 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/f3df6gxesko7s)
- [elasticsearch-periodic-platform-support #11321 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/weju75tqvkynq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.optimizer.rules.logical.HoistRemoteEnrichTopNTests.testTopNSortExpressionWithinRemoteEnrichAliasing" -Dtests.seed=BAADFF916D263D1F -Dtests.jvm.argline="-Des.concurrent_search=true" -Dtests.locale=gu -Dtests.timezone=Africa/Freetown -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.optimizer.rules.logical.HoistRemoteEnrichTopNTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testTopNSortExpressionWithinRemoteEnrichAliasing'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string starting with "$$order_by$1$0"
     but: was "$$order_by$1$8"
```

**Issue Reasons:**
- [main] 2 consecutive failures in step encryption-at-rest
- [main] 20 failures in test testTopNSortExpressionWithinRemoteEnrichAliasing (16.7% fail rate in 120 executions)
- [main] 2 failures in step encryption-at-rest (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -325,9 +325,6 @@ tests:
 - class: org.elasticsearch.xpack.search.CrossClusterAsyncSearchIT
   method: testRemoteClusterOnlyCCSWithFailuresOnAllShards
   issue: https://github.com/elastic/elasticsearch/issues/136894
-- class: org.elasticsearch.xpack.esql.optimizer.rules.logical.HoistRemoteEnrichTopNTests
-  method: testTopNSortExpressionWithinRemoteEnrichAliasing
-  issue: https://github.com/elastic/elasticsearch/issues/136957
 - class: org.elasticsearch.xpack.inference.DefaultEndPointsIT
   method: testMultipleInferencesTriggeringDownloadAndDeploy
   issue: https://github.com/elastic/elasticsearch/issues/117208

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/HoistRemoteEnrichTopNTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/HoistRemoteEnrichTopNTests.java
@@ -252,7 +252,7 @@ public class HoistRemoteEnrichTopNTests extends AbstractLogicalPlanOptimizerTest
         assertFalse(topn.local());
         Order topNOrder = topn.order().get(1);
         NamedExpression expr = as(topNOrder.child(), NamedExpression.class);
-        assertThat(expr.name(), startsWith("$$order_by$1$0"));
+        assertThat(expr.name(), startsWith("$$order_by$1$"));
         var enrich = as(topn.child(), Enrich.class);
         assertThat(enrich.mode(), is(Enrich.Mode.REMOTE));
         var innerTopN = as(enrich.child(), TopN.class);

